docs: Update our teams documentation (#235)
shorebirdtech/docs: src/content/docs/teams.mdx

to the Teams tier to add collaborators via a link in the
[Shorebird console](https://console.shorebird.dev).

## Roles

_Roles are per-app._ Not all roles are publicly editable yet. If you
have any questions about access control for your app, please do not hesitate to
reach out to contact@shorebird.dev.

### App Developer (aka Collaborator)

The Developer role can:

- Read information for a single app
- View collaborators on that app
- Create releases and patches for that app
- View insights for that app

> Developer can be added and removed by any Admin, see below.

Reviewer: ```suggestion
Developers can be added and removed by any Admin, see below.
```

### App Admin

The Admin role can:

- Everything a Developer can do

> - Add and Remove Developers for an App
> - Delete an App
> 

Reviewer: ```suggestion
- Add and Remove Developers for an App
- Delete Apps
- Delete Releases
```

Collaborators can be added and removed by any Admin, see below.
Developer can be added and removed by any Admin, see below.

### Admin (automatically assigned to the account which created the app)
### App Admin

The Admin role can:

- Everything a Developer can do
- Create and Delete Apps
- Add and Remove Collaborator for an App
- Add and Remove Developers for an App
- Delete an App

Admin permissions cannot currently be modified directly. Please email us
Admin permissions cannot currently be modified in the console. Please email us
if you need to add/remove Admins from an app.

> ### Billing account (aka App Owner)

Reviewer 2: This is a bit confusing imo because the billing account is completely orthogonal to the collaborators/roles 🤷 
Might make more sense to have billing be it's own section?

Author: I've split it out.  PTAL.

## Sharing Multiple Apps

Each app maintains a separate list of collaborators. If you need to share
multiple apps with a team, you will need to add each app separately. We plan

> to add more robust teams support later this in 2024.

Reviewer: ```suggestion
to add more robust teams support later this year.
```
